### PR TITLE
docs: add samples for pitr

### PR DESCRIPTION
**Pull request by olavloite**
Do not merge: Depends on proto changes that have not yet been released.

Adds samples for Point-in-time-recovery for Cloud Spanner.

### CODE REVIEW

**Comment by olavloite** [comment]
cc @thiagotnunes

**Comment by amanda-tarafa** [comment]
@olavloite I've turn this one into Draft and added the `do not merge` label. I'll review when it's ready, just take it out of draft and flag me.

**Review by skuruppu — APPROVED** [review]

**Review by thiagotnunes — APPROVED** [review]
LGTM

**Comment by thiagotnunes** [comment]
@olavloite Could you please change the title from `pitr-lite` to `pitr`?

**Comment by olavloite** [comment]
@amanda-tarafa This one is now ready for review. The proto changes that were needed have been released. The build failure on the Linux CI node is unrelated to this change. Would you mind taking a look?

I don't know what the process is for updating the version of the Spanner client library that is used for the samples. The three sample projects used two different versions. I've updated them all to the latest version in this PR, but I'm happy to move that to a separate PR if that's desirable.

**Comment by amanda-tarafa** [comment]
Will take a look on Monday, unless there's urgency. Thanks!

File: spanner/api/Spanner.Samples/CreateDatabaseWithRetentionPeriodAsync.cs
Comment (amanda-tarafa) [inline spanner/api/Spanner.Samples/CreateDatabaseWithRetentionPeriodAsync.cs:31]: Can you please add two variables for these two commands? It took me a while to see that `createStatement` was being added before.
Or better still, why do you need to create tables for showing how to create a DB with a retention policy? Isn't this just a littl noisy for the purpose of the sample?

File: spanner/api/Spanner.Samples/CreateDatabaseWithRetentionPeriodAsync.cs
Comment (amanda-tarafa) [inline spanner/api/Spanner.Samples/CreateDatabaseWithRetentionPeriodAsync.cs:43]: Can't the database be created from the start with a retention period?

**Review by amanda-tarafa — COMMENTED** [review]
I've left some comments/questions.

File: spanner/api/Spanner.Samples/CreateDatabaseWithRetentionPeriodAsync.cs
Comment (olavloite) [reply spanner/api/Spanner.Samples/CreateDatabaseWithRetentionPeriodAsync.cs:43]: No, unfortunately, that is not allowed by the backend.

File: spanner/api/Spanner/CreateBackup.cs
Comment (olavloite) [reply spanner/api/Spanner/CreateBackup.cs:38]: The one in `Spanner.Samples` is the one that is included in the [documentation](https://cloud.google.com/spanner/docs/samples/spanner-create-backup#spanner_create_backup-csharp), although that might be down to pure luck, as both use the same region tag.

This one is still referenced by `Program.cs`. The one in `Spanner.Samples` seems to be referenced by the integration tests.

File: spanner/api/Spanner.Samples/CreateDatabaseWithRetentionPeriodAsync.cs
Comment (olavloite) [reply spanner/api/Spanner.Samples/CreateDatabaseWithRetentionPeriodAsync.cs]: Good point, I've removed the table statements and extracted the CREATE and ALTER statements into variables.

File: spanner/api/Spanner/CreateBackup.cs
Comment (amanda-tarafa) [reply spanner/api/Spanner/CreateBackup.cs:38]: It's updated then. You need to specify both the github path and the region tag for a sample to be included.

This one will need to be cleaned up at some point. But probably we were waiting for all samples to be rewritten before cleaning up. But it's fine to include this minor change here as well.

**Review by amanda-tarafa — APPROVED** [review]
LGTM

**Comment by amanda-tarafa** [comment]
Same as before, if you are ready to merge and failures are unrelated, let me know.

**Comment by olavloite** [comment]
> Same as before, if you are ready to merge and failures are unrelated, let me know.

Yes, please go ahead and merge. The failures are unrelated to this (`VisionTest` and `ServiceAccountTests`).

**Comment by amanda-tarafa** [comment]
Yep, it's all quota related, we've been running lots of presubmits today. I'll merge after #1255.

**Pull request merged**
